project_clusters.go

```diff
@@ -0,0 +1,202 @@
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/clusters", url.QueryEscape(project))
+
+	req, err := s.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var pc []*ProjectCluster
+	resp, err := s.client.Do(req, &pc)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return pc, resp, err
+}
+
+// GetCluster gets a cluster.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/project_clusters.html#get-a-single-project-cluster
+func (s *ProjectClustersService) GetCluster(pid interface{}, cluster int, options ...OptionFunc) (*ProjectCluster, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/clusters/%d", url.QueryEscape(project), cluster)
+
+	req, err := s.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var pc *ProjectCluster
+	resp, err := s.client.Do(req, &pc)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return pc, resp, err
+}
+
+// AddClusterOptions represents the available AddCluster() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/project_clusters.html#add-existing-cluster-to-project
+type AddClusterOptions struct {
+	Name               string              `url:"name,omitempty" json:"name,omitempty"`
+	Enabled            *bool               `url:"enabled,omitempty" json:"enabled,omitempty"`
+	EnvironmentScope   string              `url:"environment_scope,omitempty" json:"environment_scope,omitempty"`
+	PlatformKubernetes *PlatformKubernetes `url:"platform_kubernetes_attributes,omitempty" json:"platform_kubernetes_attributes,omitempty"`
+}
+
+// AddCluster adds an existing cluster to the project.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/project_clusters.html#add-existing-cluster-to-project
+func (s *ProjectClustersService) AddCluster(pid interface{}, opt *AddClusterOptions, options ...OptionFunc) (*ProjectCluster, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/clusters/user", url.QueryEscape(project))
+
+	req, err := s.client.NewRequest("POST", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pc := new(ProjectCluster)
+	resp, err := s.client.Do(req, pc)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return pc, resp, err
+}
+
+// EditClusterOptions represents the available EditCluster() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/project_clusters.html#edit-project-cluster
+type EditClusterOptions struct {
+	Name               string              `url:"name,omitempty" json:"name,omitempty"`
+	EnvironmentScope   string              `url:"environment_scope,omitempty" json:"environment_scope,omitempty"`
+	PlatformKubernetes *PlatformKubernetes `url:"platform_kubernetes_attributes,omitempty" json:"platform_kubernetes_attributes,omitempty"`
+}
```

Comment:
All the fields in options structs need to be pointers.

Reply:
fixed